hack/build-integration-canary.sh

```diff
@@ -0,0 +1,438 @@
+BUILDG_CHECKSUM=buildg-v
+ROOTLESSKIT_CHECKSUM=linux
+SLIRP4NETNS_CHECKSUM=linux
+STARGZ_SNAPSHOTTER_CHECKSUM=linux
+# We specifically want the static ones
+TINI_CHECKSUM=static
+
+
+######################
+# Lib
+######################
+
+# Simple logger
+readonly LOG_LEVEL_DEBUG=0
+readonly LOG_LEVEL_INFO=1
+readonly LOG_LEVEL_WARNING=2
+readonly LOG_LEVEL_ERROR=3
+
+readonly LOG_COLOR_BLACK=0
+readonly LOG_COLOR_RED=1
+readonly LOG_COLOR_GREEN=2
+readonly LOG_COLOR_YELLOW=3
+readonly LOG_COLOR_BLUE=4
+readonly LOG_COLOR_MAGENTA=5
+readonly LOG_COLOR_CYAN=6
+readonly LOG_COLOR_WHITE=7
+readonly LOG_COLOR_DEFAULT=9
+
+readonly LOG_STYLE_DEBUG=( setaf "$LOG_COLOR_WHITE" )
+readonly LOG_STYLE_INFO=( setaf "$LOG_COLOR_GREEN" )
+readonly LOG_STYLE_WARNING=( setaf "$LOG_COLOR_YELLOW" )
+readonly LOG_STYLE_ERROR=( setaf "$LOG_COLOR_RED" )
+
+_log::log(){
+  local level
+  local style
+  local numeric_level
+  local message="$2"
+
+  level="$(printf "%s" "$1" | tr '[:lower:]' '[:upper:]')"
+  numeric_level="$(printf "LOG_LEVEL_%s" "$level")"
+  style="LOG_STYLE_${level}[@]"
+
+  [ "${!numeric_level}" -ge "$LOG_LEVEL" ] || return 0
+
+  [ ! "$TERM" ] || [ ! -t 2 ] || >&2 tput "${!style}" 2>/dev/null || true
+  >&2 printf "[%s] %s: %s\n" "$(date 2>/dev/null || true)" "$(printf "%s" "$level" | tr '[:lower:]' '[:upper:]')" "$message"
+  [ ! "$TERM" ] || [ ! -t 2 ] || >&2 tput op 2>/dev/null || true
+}
+
+log::init(){
+  # Default log to warn if unspecified
+  _ll="$(printf "LOG_LEVEL_%s" "${NERDCTL_CI_LOG_LEVEL:-warn}" | tr '[:lower:]' '[:upper:]')"
+  # Default to 3 (warn) if unrecognized
+  LOG_LEVEL="${!_ll:-3}"
+}
+
+log::debug(){
+  _log::log debug "$@"
+}
+
+log::info(){
+  _log::log info "$@"
+}
+
+log::warn(){
+  _log::log warning "$@"
+}
+
+log::error(){
+  _log::log error "$@"
+}
+
+# Helpers
+host::require(){
+  local binary="$1"
+  command -v "$binary" >/dev/null || {
+    log::error "You need $binary for this script to work, and it cannot be found in your path"
+    exit 1
+  }
+}
+
+fs::mktemp(){
+  mktemp -dq "${TMPDIR:-/tmp}/$prefix.XXXXXX" 2>/dev/null || mktemp -dq || {
+    log::error "Failed to create temporary directory"
+    exit 1
+  }
+}
+
+http::get(){
+  local args=(curl --proto '=https' --tlsv1.2 -fsSL)
+  args+=("$@")
+
+  log::debug "${args[*]}"
+  "${args[@]}"
+}
+
+http::checksum(){
+  local urls=("$@")
+  local url
+
+  local prefix="nerdctl-checksum"
+
+  local temp
+  temp="$(fs::mktemp)"
+
+  for url in "${urls[@]}"; do
+    http::get -o "$temp/${url##*/}" "$url"
+  done
+
+  cd "$temp"
+  shasum -a 256 ./*
+  cd - >/dev/null || true
+}
+
+# Github API helpers
+# Set NERDCTL_GITHUB_API_TOKEN to use authenticated requests to workaround limitations
+github::request(){
+  local endpoint="$1"
+  local args=(
+    -H "Accept: application/vnd.github+json"
+    -H "X-GitHub-Api-Version: 2022-11-28"
+  )
+
+  [ "${NERDCTL_GITHUB_API_TOKEN:-}" == "" ] || args+=(-H "Authorization: Bearer $NERDCTL_GITHUB_API_TOKEN")
```

Comment:
`NERDCTL_` prefix might not be needed?

Reply:
> `NERDCTL_` prefix might not be needed?

Yeah I was thinking about that. Might be more convenient for peeps. On the other hand, for these who like to segregate tokens, it is convenient to be able to use a different name.

Will look into this again tomorrow on a laptop

Reply:
Ok, renamed to just GITHUB_TOKEN and hooked-in the `secret` auto generated thing (hope that works as I think it does...).